File: vane-regions/src/main/java/org/oddlama/vane/regions/commands/Region.java
package org.oddlama.vane.regions.commands;

import static com.mojang.brigadier.Command.SINGLE_SUCCESS;

import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import io.papermc.paper.command.brigadier.CommandSourceStack;
import org.bukkit.entity.Player;
import org.oddlama.vane.annotation.command.Aliases;
import org.oddlama.vane.annotation.command.Name;
import org.oddlama.vane.core.command.Command;
import org.oddlama.vane.core.module.Context;
import org.oddlama.vane.regions.Regions;

@Name("region")
@Aliases({ "regions", "rg" })
public class Region extends Command<Regions> {

    public Region(Context<Regions> context) {
        super(context);
    }

    @Override
    public LiteralArgumentBuilder<CommandSourceStack> getCommandBase() {
        return super.getCommandBase()
            .then(help())
            .requires(ctx -> ctx.getSender() instanceof Player)
            .executes(ctx -> {
                openMenu((Player) ctx.getSource().getSender());
                return SINGLE_SUCCESS;
            });
    }

    private void openMenu(Player player) {
        getModule().menus.mainMenu.create(player).open(player);
    }
}

File: vane-regions/src/main/kotlin/org/oddlama/vane/regions/commands/Region.kt
package org.oddlama.vane.regions.commands

import com.mojang.brigadier.Command
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext
import io.papermc.paper.command.brigadier.CommandSourceStack
import org.bukkit.entity.Player
import org.oddlama.vane.annotation.command.Aliases
import org.oddlama.vane.annotation.command.Name
import org.oddlama.vane.core.module.Context
import org.oddlama.vane.regions.Regions

@Name("region")
@Aliases("regions", "rg")
/**
 * Root command opening the regions management menu.
 */
class Region(context: Context<Regions?>) : org.oddlama.vane.core.command.Command<Regions?>(context) {
    /**
     * Builds the `/region` command tree.
     */
    override fun getCommandBase(): LiteralArgumentBuilder<CommandSourceStack> {
        return super.getCommandBase()
            .then(help())
            .requires { source: CommandSourceStack -> source.sender is Player }
            .executes { ctx: CommandContext<CommandSourceStack> ->
                openMenu(ctx.source.sender as Player)
                Command.SINGLE_SUCCESS
            }
    }

    /**
     * Opens the main regions menu for the given player.
     */
    private fun openMenu(player: Player) {
        module?.menus?.mainMenu?.create(player)?.open(player)
    }
}
